src/main/java/com/wix/mysql/distribution/service/CatchAllCommandEmitter.java

```diff
@@ -31,10 +31,28 @@ public boolean matches(Version version) {
                 format("--plugin-dir=%s/lib/plugin", baseDir),
                 format("--lc-messages-dir=%s/share", baseDir),
                 format("--port=%s", config.getPort()),
+                format("--socket=%s", sockFile(exe)),
                 "--console",
                 format("--character-set-server=%s", config.getCharset().getCharset()),
                 format("--collation-server=%s", config.getCharset().getCollate()),
                 format("--default-time-zone=%s", Utils.asHHmmOffset(config.getTimeZone())));
 
     }
+
+    /**
+     * Helper for getting stable sock classPathFile. Saving to local instance variable on service start does not work due
+     * to the way flapdoodle process library works - it does all init in {@link de.flapdoodle.embed.process.runtime.AbstractProcess} and instance of
+     * {@link com.wix.mysql.MysqldProcess} is not yet present, so vars are not initialized.
+     * This algo gives stable sock classPathFile based on single executeCommands profile, but can leave trash sock classPathFiles in tmp dir.
+     * <p>
+     * Notes:
+     * .sock classPathFile needs to be in system temp dir and not in ex. target/...
+     * This is due to possible problems with existing mysql installation and apparmor profiles
+     * in linuxes.
+     */
+    private String sockFile(IExtractedFileSet exe) throws IOException {
```

Comment:
Use [java.nio.Files](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html) for temp file creation